### PR TITLE
Fix/issue-3313 [Reports][Admin Panel] USD amount isn't automatically cleared when clearing UAH amount field

### PR DESCRIPTION
**Pull request by piprock**
## Github ticket

* [Github ticket](https://github.com/ita-social-projects/VictoryCenter-Back/issues/3313)

## Description

This PR resolves an issue where clearing the 'Сума (UAH)' input field in the funds expenditures record modal did not automatically clear the converted value in the 'Сума (USD)' field. 

Previously, clearing the UAH input left the stale converted USD value behind because conversion logic only ran when non-empty inputs passed validation.

## How it looks

https://github.com/user-attachments/assets/07336e9c-b3ad-47de-888d-306bb156417c

## Summary of change

* **Auto-clearing USD amount (`updateFundsAmounts`)**: Updated `updateFundsAmounts` logic to explicitly reset `nextAmountUsd` to an empty string (`''`) and clear `nextAmountUsdError` whenever the normalized UAH input value is empty (`normalized === ''`).
* **Unit Tests (`updateFundsAmounts.test.ts`)**: Added unit tests covering USD amount/error clearing logic for both `change` and `blur` triggers when the UAH input becomes empty.

## How to recreate changes

1. Log in as an Admin and navigate to **"Звітність"** -> **"Звіт та аналітика"** -> **"Доходи та витрати"**.
2. Click **"Редагувати Доходи та витрати"** navigate to **"Програмні витрати"** tab and click **"+ Витрата по програмі"**.
3. Enter a numeric value in the **"Сума (UAH)"** field (e.g., `120`).
4. Verify that **"Сума (USD)"** is automatically populated with the converted value.
5. Completely clear the value in the **"Сума (UAH)"** field.
6. Observe that the **"Сума (USD)"** field is now automatically cleared as well, leaving both amount fields empty.

## CHECK LIST
- [x] New tests was added or existing was modified
- [ ] Changes has severe impact on users
- [ ] Changes has medium impact on users
- [x] Changes has light impact on users
- [ ] Test coverage was updated
- [x] PR meets all conventions

## Summary by CodeRabbit

* **Bug Fixes**
  * Clearing the UAH amount now also clears the converted USD amount and its validation message.
  * USD conversion only occurs when a valid, non-empty UAH amount is provided.
  * Converted USD values continue to receive validation checks.

### CODE REVIEW

**Review by stkossman — APPROVED** [review]

**Review by cirillad — APPROVED** [review]

File: src/utils/functions/update-funds-amounts/update-funds-amounts.ts [hunk 1]
```diff
@@ -29,12 +29,17 @@ export const updateFundsAmounts = (
...
+            } else if (!currentFieldError) {
+                const convertedAmount = getConvertedAmount(normalized, exchangeRate);
 
-            if (convertedAmount !== null) {
-                nextAmountUsd = convertedAmount;
-                nextAmountUsdError = validateFundsExpendituresAmount(convertedAmount, trigger);
+                if (convertedAmount !== null) {
+                    nextAmountUsd = convertedAmount;
+                    nextAmountUsdError = validateFundsExpendituresAmount(convertedAmount, trigger);
+                }
```
Comment (mehalyna) [inline src/utils/functions/update-funds-amounts/update-funds-amounts.ts:44]: The condition logic could be simplified and made more readable. The nested if-else structure makes it harder to track the flow - consider moving the empty check outside the error validation to reduce cognitive load and improve clarity.

File: src/utils/functions/update-funds-amounts/update-funds-amounts.test.ts [hunk 2]
```diff
@@ -268,6 +268,39 @@ describe('updateFundsAmounts', () => {
             expect(result.amountUah).toBe('');
         });
 
+        it.each([
+            ['change', undefined],
+            ['blur', 'Field is required'],
+        ] as const)('should clear amountUsd when amountUah becomes empty on %s trigger', (trigger, uahError) => {
```
Comment (mehalyna) [inline src/utils/functions/update-funds-amounts/update-funds-amounts.test.ts:274]: The test title references both 'change' and 'blur' triggers but the parameterized test only verifies state clearing logic. Add an assertion that validates the exchange rate conversion is skipped when amountUah is empty (e.g., expect(mockGetConvertedAmount).not.toHaveBeenCalled()).

File: src/utils/functions/update-funds-amounts/update-funds-amounts.test.ts [hunk 3]
```diff
@@ -268,6 +268,39 @@ describe('updateFundsAmounts', () => {
...
+            mockNormalizeFundsExpendituresAmountInput.mockReturnValue('');
+            mockValidateFundsExpendituresAmount.mockImplementation(
+                (value: string, currentTrigger?: 'change' | 'blur' | 'save') => {
+                    if (value === '') {
+                        return currentTrigger === 'blur' ? 'Field is required' : undefined;
+                    }
+
+                    return undefined;
+                },
+            );
```
Comment (mehalyna) [inline src/utils/functions/update-funds-amounts/update-funds-amounts.test.ts:293]: The mock implementation for validateFundsExpendituresAmount only handles the empty string case and always returns undefined for non-empty values. Add a test case where a non-empty UAH value produces a validation error to ensure error handling is still applied after clearing.

**Review by mehalyna — APPROVED** [review]
